pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -65,6 +66,10 @@ func (r *Cluster) ValidateCreate() error {
 		}
 	}
 
+	if err := ValidateControlPlaneIPUniqueness(r, &networkutils.DefaultNetClient{}); err != nil {
```

Comment:
This will result in at least a 2.5s lag period for every submission; that's rather significant. Are we content with that? (Webhooks default timeout is 10s).

Reply:
Hmm I'm not sure of how we can avoid that, but I'm open to ideas for sure.

Without having the validation here, creating a cluster using the new API with a control plane IP that's already in use will result in it continuing on with cluster creation and getting stuck instead of validating beforehand that the ip wont be assigned